components/DependencyPropertyGenerator/CommunityToolkit.DependencyPropertyGenerator.Tests/Test_DependencyPropertyGenerator.cs

```diff
@@ -0,0 +1,1962 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using CommunityToolkit.GeneratedDependencyProperty.Tests.Helpers;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace CommunityToolkit.GeneratedDependencyProperty.Tests;
+
+[TestClass]
+public partial class Test_DependencyPropertyGenerator
+{
+    [TestMethod]
+    public void SingleProperty_Int32_WithLocalCache()
+    {
+        const string source = """
+            using Windows.UI.Xaml;
+            using CommunityToolkit.WinUI;
+
+            namespace MyNamespace;
+
+            public partial class MyControl : DependencyObject
+            {
+                [GeneratedDependencyProperty(IsLocalCacheEnabled = true)]
+                public partial int Number { get; set; }
+            }
+            """;
+
+        const string result = """
+            // <auto-generated/>
+            #pragma warning disable
+            #nullable enable
+
+            namespace MyNamespace
+            {
+                /// <inheritdoc cref="MyControl"/>
+                partial class MyControl
+                {
+                    /// <summary>
+                    /// The backing <see cref="global::Windows.UI.Xaml.DependencyProperty"/> instance for <see cref="Number"/>.
+                    /// </summary>
+                    [global::System.CodeDom.Compiler.GeneratedCode("CommunityToolkit.WinUI.DependencyPropertyGenerator", <ASSEMBLY_VERSION>)]
+                    public static readonly global::Windows.UI.Xaml.DependencyProperty NumberProperty = global::Windows.UI.Xaml.DependencyProperty.Register(
+                        name: "Number",
+                        propertyType: typeof(int),
+                        ownerType: typeof(MyControl),
+                        typeMetadata: new global::Windows.UI.Xaml.PropertyMetadata(default(int)));
+
+                    /// <inheritdoc/>
+                    [global::System.CodeDom.Compiler.GeneratedCode("CommunityToolkit.WinUI.DependencyPropertyGenerator", <ASSEMBLY_VERSION>)]
+                    [global::System.Diagnostics.DebuggerNonUserCode]
+                    [global::System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
+                    public partial int Number
+                    {
+                        get => field;
```

Comment:
We should be calling `GetValue` eh and not using the `field` here? https://learn.microsoft.com/en-us/windows/uwp/xaml-platform/custom-dependency-properties#implementing-the-wrapper

We should also note that the methods aren't guaranteed to be called in the setter either.